### ISSUE TITLE
Small bug in Document/Models/Project.swift

### ISSUE THREAD

**Issue by wjk** [ChimeHQ/Chime#2]
While reading through your code I noticed the following:

https://github.com/ChimeHQ/Chime/blob/72e6297f6971a2dd235701016390267584fe60e5/Edit/Modules/Document/Models/Project.swift#L72-L73

If Chime is built in release, the call to `textDocuments.remove` will be elided, as it is only used in the context of the assert call, and those are removed when doing a release build. As a result, you wouldn’t be able to successfully remove documents from projects. You would need to do this to make it work:

```swift
case let doc as TextDocument:
    let removed: Bool = textDocuments.remove(doc) != nil
    assert(removed)
```

**Comment by mattmassicotte** [comment]
Oh boy, that isn't the first time I've made a mistake like that. Thank you so much!

**Comment by mattmassicotte** [comment]
Fixed with 0a00ba614e7ff04c75ba4a8b890fc2200494b213
